### ISSUE TITLE
Numpy: Root Path Attributes

### ISSUE THREAD

**Issue by ax3l** [ornladios/ADIOS#78]
@yyalli For an example file such as
```bash
$ bpls -Ad test.bp
  long long  a1    attr   = 12
  long long  /a2   attr   = 42
```

reading attributes relative to the base path is not always working
```python
import adios as ad
f = ad.File("test.bp")
f.attrs["a1"] # works
f.attrs["/a1"] # fails
f.attrs["a2"] # fails
f.attrs["/a2"] # works
f.close()
```

But access wise, reading with our without the `/` in the File object should be identical.

**Comment by yyalli** [comment]
Reading with and without '/' provided in the File object is an extra convenience. As you mentioned, the File object can handle as follows:
f["a1"] # works
f["/a1"] # works
f["a2"] # works
f["/a2"] # works

I think it doesn't need to be identical with attr objects which represent "attr" objects in the file. How do you think?

On Jul 17, 2016, at 5:56 PM, Axel Huebl <notifications@github.com<mailto:notifications@github.com>> wrote:


@yyalli<https://github.com/yyalli> For an example file such as

$ bpls -Ad test.bp
  long long  a1    attr   = 12
  long long  /a2   attr   = 42

reading attributes relative to the base path is not always working

import adios as ad
f = ad.File("test.bp")
f.attrs["a1"] # works
f.attrs["/a1"] # fails
f.attrs["a2"] # fails
f.attrs["/a2"] # works
f.close()

But access wise, reading with our without the / in the File object should be identical.

**Comment by ax3l** [comment]
Yes, what I suggest is that it should work not only for vars such as f["v1"]==f["/v1"] but also for attributes f.attrs["a1"]==f.attrs["/a1"]

**Comment by yyalli** [comment]
I have made changes and it should work as you describe. Please let me know if you find any problem.

On Jul 18, 2016, at 6:09 PM, Axel Huebl <notifications@github.com<mailto:notifications@github.com>> wrote:

Yes, what I suggest is that it should work not only for vars such as f["v1"]==f["/v1"] but also for attributes f.attrs["a1"]==f.attrs["/a1"]
